larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+
+inline float DiscreteProbabilityVector::GetWidth(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetWidth();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void DiscreteProbabilityVector::GetAllAtIndex(const size_t index, float &x, float &probabilityDensity,
+    float &cumulativeProbability, float &width) const
+{
+    this->VerifyElementRequest(index);
+
+    const DiscreteProbabilityDatum &theDatum(m_discreteProbabilityData.at(index));
+    x = theDatum.GetX();
+    probabilityDensity = theDatum.GetDensityDatum();
+    cumulativeProbability = theDatum.GetCumulativeDatum();
+    width = theDatum.GetWidth();
+
+    return;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline DiscreteProbabilityVector::DiscreteProbabilityDatum::DiscreteProbabilityDatum(const float x,
+    const float densityDatum, const float cumulativeDatum, const float width) :
+    m_x(x),
+    m_densityDatum(densityDatum),
+    m_cumulativeDatum(cumulativeDatum),
+    m_width(width)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::DiscreteProbabilityDatum::GetX() const
+{
+    return m_x;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::DiscreteProbabilityDatum::GetDensityDatum() const
+{
+    return m_densityDatum;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::DiscreteProbabilityDatum::GetCumulativeDatum() const
+{
+    return m_cumulativeDatum;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::DiscreteProbabilityDatum::GetWidth() const
+{
+    return m_width;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void DiscreteProbabilityVector::VerifyCompleteData() const
+{
+    if (2 > m_discreteProbabilityData.size())
+        throw pandora::StatusCodeException(pandora::STATUS_CODE_NOT_INITIALIZED);
+
+    if (m_discreteProbabilityData.back().GetX() - m_xUpperBound > std::numeric_limits<float>::epsilon())
+        throw pandora::StatusCodeException(pandora::STATUS_CODE_INVALID_PARAMETER);
+    return;
```

Comment:
Can see why you might want to explicitly return here. If so, typically have empty line after if statement.

Reply:
Space added!